### ISSUE TITLE
ModSettings.Remove seems broken

### ISSUE THREAD

**Issue by DorianBourgeoisat** [rowlul/modthara#8]
Hullo,
I tried to write an empty modsettings by removing each mod one by one from the document, using the Remove method in ModSettings.cs.
Ofc there are much cleaner approaches, but I wanted to familiarise myself with the codebase.
```cs
while (_modSettingsService.ModSettings.Mods.Count > 0)
{
    _modSettingsService.ModSettings.Remove(_modSettingsService.ModSettings.Mods[0]);
}
var l = new List<Module>(_modSettingsService.ModSettings.Mods);
foreach (var m in l) {
    Console.WriteLine("Remaining module " + m.Name + " UUID " + m.Uuid.ToString());
}
Console.WriteLine("WRITING");
StreamReader r = new StreamReader(_modSettingsService.ModSettings.ToDocument().ToStream());
string docu = r.ReadToEnd();
Console.WriteLine(docu);
```
Such code leads to a seemingly empty mod list, but the docu string is NOT emptied, in fact, it is identical to the initial document.
I went to the ModSettings.cs file and checked for an inconsistency : 
```cs
public void Remove(Module mod)
    {
        int mocc = ModOrderChildren.Count;
        int mcc = ModsChildren.Count;
        int mdsc = _mods.Count;

        ModOrderChildren.Remove(((ModuleBase)mod).ToNode());
        ModsChildren.Remove(mod.ToNode());
        _mods.Remove(mod);

        int moccp = ModOrderChildren.Count;
        int mccp = ModsChildren.Count;
        int mdscp = _mods.Count;

        if(mdscp < mdsc)
        {
            if((mccp >= mcc) || (moccp >= mocc))
            {
                throw new Exception("Not removed");
            }
        }
    }
```
And sure enough : 
```
sulla% dotnet run --project Modthara.UI/Modthara.UI.csproj
Unhandled exception. System.Exception: Not removed
```
So I want to know if there's a subtlety to using the ModSettings.cs methods that I didn't see of if it is indeed a bug that should be further investigated.

**Comment by rolowilde** [comment]
It is indeed a bug since ModuleBase and LsxNode do not implement IEquatable<T>. I'm about to fix this oversight of mine.

Edit: there's actually no need to implement it since mod index in the list is easily obtainable

**Comment by rowlul** [comment]
Also, like List<T>.Remove() it should return a bool:
> true if item is successfully removed; otherwise, false. This method also returns false if item was not found in the List<T>.

**Comment by rolowilde** [comment]
Fixed in 9c035778adcc96ed0d89a2a323ec2d21695a9d7b.
